Admin.js

```diff
@@ -114,561 +136,771 @@ var Admin = {
+                    jQuery("tbody input[type=\"checkbox\"]", subject).each((i, e) => {
+
                         if (i > previousIndex && i < currentIndex || i > currentIndex && i < previousIndex) {
+
                             if (useICheck) {
-                                jQuery(e).iCheck(isChecked ? 'check' : 'uncheck');
+
+                                jQuery(e).iCheck(isChecked ? "check" : "uncheck");
 
                                 return;
+
                             }
 
-                            jQuery(e).prop('checked', isChecked);
+                            jQuery(e).prop("checked", isChecked);
+
                         }
+
                     });
+
                 }
 
-                previousIndex  = currentIndex;
+                previousIndex = currentIndex;
+
             }
+
         });
+
     },
 
-    setup_xeditable: function(subject) {
-        Admin.log('[core|setup_xeditable] configure xeditable on', subject);
-        jQuery('.x-editable', subject).editable({
-            emptyclass: 'editable-empty btn btn-sm btn-default',
-            emptytext: '<i class="fa fa-pencil"></i>',
-            container: 'body',
-            placement: 'auto',
-            success: function(response) {
-                var html = jQuery(response);
+    "setup_xeditable" (subject) {
+
+        Admin.log("[core|setup_xeditable] configure xeditable on", subject);
+        jQuery(".x-editable", subject).editable({
+            "emptyclass": "editable-empty btn btn-sm btn-default",
+            "emptytext": "<i class=\"fa fa-pencil\"></i>",
+            "container": "body",
+            "placement": "auto",
+            "success" (response) {
+
+                const html = jQuery(response);
                 Admin.setup_xeditable(html);
-                jQuery(this)
-                    .closest('td')
-                    .replaceWith(html);
+                jQuery(this).
+                    closest("td").
+                    replaceWith(html);
+
             },
-            error: function(xhr, statusText, errorThrown) {
+            "error" (xhr, statusText, errorThrown) {
+
                 return xhr.responseText;
+
             }
         });
+
     },
 
     /**
-     * render log message
+     * Render log message
      * @param mixed
      */
-    log: function() {
-        var msg = '[Sonata.Admin] ' + Array.prototype.join.call(arguments,', ');
+    "log" () {
+
+        const msg = `[Sonata.Admin] ${Array.prototype.join.call(arguments, ", ")}`;
         if (window.console && window.console.log) {
+
             window.console.log(msg);
+
         } else if (window.opera && window.opera.postError) {
+
             window.opera.postError(msg);
+
         }
+
     },
 
     /**
      * NEXT_MAJOR: remove this function.
      *
      * @deprecated in version 3.0
      */
-    add_pretty_errors: function() {
-        console.warn('Admin.add_pretty_errors() was deprecated in version 3.0');
+    "add_pretty_errors" () {
```

Comment:
- eslint: Expected object keys to be in ascending order. 'add_pretty_errors' should be before 'log'.